### PR TITLE
KTLN-706: Create a Frequency Map in Kotlin

### PR DESCRIPTION
**Pull request by nappy29**
added unit tests

### CODE REVIEW

File: core-kotlin-modules/core-kotlin-datastructures/src/test/kotlin/com/baeldung/frequencymap/FrequencyMapUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    return map
+}
+
+fun frequencyMapUsingBinarySearchMethod(list: List<Int>): MutableMap<Int, Int>{
+    val sortedList = list.sorted()
+
+    val map = mutableMapOf<Int, Int>()
+
+    sortedList.forEach { element ->
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-datastructures/src/test/kotlin/com/baeldung/frequencymap/FrequencyMapUnitTest.kt:97]: One small optimization here:

```suggestion
    sortedList.distinct().forEach { element ->
```

Comment (nappy29) [reply core-kotlin-modules/core-kotlin-datastructures/src/test/kotlin/com/baeldung/frequencymap/FrequencyMapUnitTest.kt]: Thanks for pointing that out.
Fixed!

File: core-kotlin-modules/core-kotlin-datastructures/src/test/kotlin/com/baeldung/frequencymap/FrequencyMapUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import org.junit.Test
+import java.util.*
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+
+class FrequencyMapUnitTest {
+
+    @Test
+    fun `test frequency map using mutable map method`() {
+        val list = listOf(1, 2, 1, 3, 2, 4, 4, 7,9,7,3,2,1)
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-datastructures/src/test/kotlin/com/baeldung/frequencymap/FrequencyMapUnitTest.kt:12]:
```suggestion
        val list = listOf(1, 2, 1, 3, 2, 4, 4, 7, 9, 7, 3, 2, 1)
```

Comment (KevinGilmore) [reply core-kotlin-modules/core-kotlin-datastructures/src/test/kotlin/com/baeldung/frequencymap/FrequencyMapUnitTest.kt:12]: We need to add spaces here for readability. Let's apply this change in the other tests, too.

**Pull request closed without merging**